### PR TITLE
#23/UI/leaderboard

### PR DESCRIPTION
**Pull request by RobertS-ki**
Updated 25.01.2022:
-code refactored
-splip logic into separate function files
-updated styles

Updated 24.01.2022:
-code refactored,
-moved files,
-updated styles
obraz
===== Previous:
obraz
Component - molecule, consists of 3 atoms (for the time beeing).
obraz
Accepts 3 pairs of [key: value] via props:
obraz

### CODE REVIEW

File: src/components/Leaderboard.js
Comment (vieraboschkova) [inline src/components/Leaderboard.js:15]: See if you could destructure props

File: src/components/Leaderboard.js
Comment (vieraboschkova) [inline src/components/Leaderboard.js:21]: Very good question! I think you might find an answer here: https://pl.reactjs.org/docs/lists-and-keys.html#rendering-multiple-components

File: src/components/Leaderboard.js
Comment (vieraboschkova) [inline src/components/Leaderboard.js:17]: Not sure if this needs to be called a molecule, maybe a naming convention would be something you could discuss with the team?

File: src/components/LeaderboardBtn.js
Comment (vieraboschkova) [inline src/components/LeaderboardBtn.js:3]: Is this going to be a button? I thought it was a table row, is there any functionality planned? Maybe a link within the table would be enough if you are planning to add some redirecting?

Comment (MonikaKrella) [inline src/components/LeaderboardBtn.js:3]: Maybe also put styles to another file? I'll make discussion at group and then file with some tips

**Comment by MonikaKrella** [comment]
there could be also changes at Figma in leaderboard, look at Figma soon or ask Ala

File: src/components/LeaderboardBtn.js
Comment (vieraboschkova) [inline src/components/LeaderboardBtn.js:32]: Perhaps you could think of another tag to use in here?

File: src/components/LeaderboardBtn.js
Comment (KatarzynaChmielecka) [inline src/components/LeaderboardBtn.js:8]: Now you can use color vars. You find them in src/style folder.

**Review by MonikaKrella — COMMENTED** [review]
image
image

when changed to disqualification, leaderboard is white - at Figma should be grey

File: src/Atoms/Leaderboard/SummaryLine.js
Comment (MonikaKrella) [inline src/Atoms/Leaderboard/SummaryLine.js:17]: image
should be "punkty"?

File: src/Molecules/LeaderboardList/LeaderboardList.js
Comment (MonikaKrella) [inline src/Molecules/LeaderboardList/LeaderboardList.js:22]: should be required?

File: src/Pages/SingleSummaryPage/SingleSummaryPage.js
Comment (MonikaKrella) [inline src/Pages/SingleSummaryPage/SingleSummaryPage.js:15]: maybe it will be faster and less logic lower, when in that page will be countig sum of points and then sending by props. And whole array just where it is needed?

File: src/Atoms/Leaderboard/SummaryLine.js
Comment (RobertS-ki) [reply src/Atoms/Leaderboard/SummaryLine.js:17]: Nice - you tested different values! Its a result of "change all occurences" in use - the function was named "punkty" before :)

File: src/Atoms/Leaderboard/LeaderboardListElement.js
Comment (RobertS-ki) [reply src/Atoms/Leaderboard/LeaderboardListElement.js:8]: Fixed.

File: src/Atoms/Leaderboard/penaltyScoreCalc.js
Comment (RobertS-ki) [reply src/Atoms/Leaderboard/penaltyScoreCalc.js:1]: Done.

File: src/Molecules/LeaderboardList/LeaderboardList.js
Comment (RobertS-ki) [reply src/Molecules/LeaderboardList/LeaderboardList.js:22]: Nothing will run without the object input, so yes :)

File: src/Pages/SingleSummaryPage/SingleSummaryPage.js
Comment (RobertS-ki) [reply src/Pages/SingleSummaryPage/SingleSummaryPage.js:15]: Could be, but the page itself is not the part of the scope (?)
It seems only one ModalPage has some more logic ony the page.
Maybe we should decide in a team?

**Review by MonikaKrella — COMMENTED** [review]
image
-disqualification mode looks good
-but now yellow card doesn't work:
image

**Review by MonikaKrella — COMMENTED** [review]
good work!
after merge the latest development will be approved

**Review by MonikaKrella — APPROVED** [review]
works after merge development, good work!

**Pull request merged**
